usr/sbin/registercloudguest

```diff
@@ -95,11 +96,80 @@ def get_register_cmd():
     return register_cmd
 
 
+# ----------------------------------------------------------------------------
+def run_SUSEConnect(
+    registration_target, regcode='', email='',
+    instance_data_filepath='', product=''
+):
+    """
+    Wrapper for SUSEConnect
+
+    Call SUSEConnect or the respective transactional update call
+    with the given parameters:
+
+    registration_target:
+        SMT registration object
+    regcode:
+        registration code
+    email:
+        registration e-mail
+    instance_data_filepath:
+        CSP related metadata file
+    product:
+        SUSE product triplet
+    """
+    suseconnect_type = namedtuple(
+        'suseconnect_type', ['returncode', 'output', 'error']
+    )
+    register_cmd = get_register_cmd()
+    if not (os.path.exists(register_cmd) and os.access(register_cmd, os.X_OK)):
+        logging.error('No registration executable found')
+        print(err_msg, file=sys.stderr)
+        sys.exit(1)
+
+    cmd = [register_cmd]
+    target_url = registration_target.get_FQDN()
+
+    # distinguish command between standard and read-only system(transactional)
+    if 'transactional' in register_cmd:
+        cmd += ['register', '--url', 'https://{0}'.format(target_url)]
+    else:
+        cmd += ['--url', 'https://{0}'.format(target_url)]
+
+    # SUSE product triplet
+    if product:
+        cmd += ['--product', product]
+
+    # set path to a metadata file, cloud specific
+    if instance_data_filepath:
+        cmd += ['--instance-data', instance_data_filepath]
+
+    # set e-mail
+    if email:
+        cmd += ['--email', email]
+
+    # set registration code for given product
+    if regcode:
+        cmd += ['--regcode', regcode]
+
+    logging.info('Registration: {0}'.format(' '.join(cmd)))
+    call = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
```

Comment:
@Johannes should we, do we need to sanitize the log, i.e. do not log the registration code?

Reply:
Funny, I am working on Python code right now using Popen and doing something not too different, but still you highlighted the wrong person :)

Good luck with your project!

Reply:
Funny, sorry @johannes couldn't find the Johannes I was looking for and was only going for the name without realizing there was a user reference just for Johannes.

Reply:
Per Slack conversation with another Johannes, we should not log the reg code. @schaefi please update accordingly. Thanks

Reply:
All right, I will adapt the PR

Reply:
done